syntax = "proto3";
package PythonService;

option go_package = "github.com/arnavsharma/central/gen/go/pythonservicepb";
service PythonUtil {
  rpc ParseNewsArticle(ParseNewsArticleRequest) returns (ParseNewsArticleResponse);
  rpc GenerateEmbeddings(GenerateEmbeddingsRequest) returns (GenerateEmbeddingsResponse);
  rpc GenerateKeywords(GenerateKeywordsRequest) returns (GenerateKeywordsResponse);
}
message ParseNewsArticleRequest {
  string url = 1;
}
message ParseNewsArticleResponse {
  string provider = 1;
  string metadata = 2;
  string headline = 3;
  string content = 4;
}
message GenerateEmbeddingsRequest {
  string content = 1;
}
message GenerateEmbeddingsResponse {
  repeated float embeddings = 1;
}
message GenerateKeywordsRequest {
  string content = 1;
}

message GenerateKeywordsResponse {
  repeated string keywords = 1;
}
